pkg/grpc_middleware/dialer/config.go

```diff
@@ -0,0 +1,196 @@
+package dialer
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/caring/go-packages/v2/pkg/errors"
+)
+
+// Config read from environment or connection builder.
+type Config struct {
+	host       string
+	port       string
+	caFile     string
+	clientCert string
+	clientKey  string
+	tokenAuth  string
+	basicAuth  string
+	withTLS    bool
+	skipVerify bool
+}
+
+// ReadConfig reads connection string from provided env variable name and returns config.
+// If there are tls options the Config value can be used to generate the tls.Config
+// the format is tls://hostname:port (or tcp://hostname:port if developing locally without TLS)
+// Optional TLS parameters:
+//   skip_verify=true        ignore server CA verification.
+//   ca_file=./filename.pem  CA of service for verification.
+// NOTE: The connection process is non-blocking so a timeout is not needed. To force a blocking add grpc.Blocking() to the opts
+func ReadConfig(addr string) (*Config, error) {
+	if addr == "" {
+		return nil, errors.Errorf("connection string is empty")
+	}
+
+	var c Config
+	// If there is a connection string use that.
+	// format: tls://hostname:port or tcp://hostname:port
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, errors.Wrap(err, "connection string is malformed")
+	}
+
+	c.host = u.Hostname()
+	switch u.Scheme {
```

Comment:
Thoughts on adding a case here to default to tls when there is no scheme? This would make it more compatible with current `host + port` setups.